src/ui/layout/tutorial.md

```diff
@@ -244,219 +393,287 @@ class MyApp extends StatelessWidget {
+1. Call the helper method three times, once for each button.
+1. Pass the color, `Icon`, and text for that specific button.
+1. Align the columns along the main axis with the
+   `MainAxisAlignment.spaceEvenly` method.
+   This arranges the free space in equal amounts before, between,
+   and after each column.
+
+<?code-excerpt "lib/main.dart (ButtonSection)" title?>
 ```dart
-Color color = Theme.of(context).primaryColor;
-
-Widget buttonSection = Row(
-  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-  children: [
-    _buildButtonColumn(color, Icons.call, 'CALL'),
-    _buildButtonColumn(color, Icons.near_me, 'ROUTE'),
-    _buildButtonColumn(color, Icons.share, 'SHARE'),
-  ],
-);
-```
+class ButtonSection extends StatelessWidget {
+  const ButtonSection({super.key});
 
-Add the button section to the body:
+  @override
+  Widget build(BuildContext context) {
+    final Color color = Theme.of(context).primaryColor;
+    return SizedBox(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: [
+          BuildButtonColumn(
+            color: color,
+            icon: Icons.call,
+            label: 'CALL',
+          ),
+          BuildButtonColumn(
+            color: color,
+            icon: Icons.near_me,
+            label: 'ROUTE',
+          ),
+          BuildButtonColumn(
+            color: color,
+            icon: Icons.share,
+            label: 'SHARE',
+          ),
+        ],
+      ),
+    );
+  }
+}
 
-<?code-excerpt path-base="layout/lakes"?>
-<?code-excerpt "{step2,step3}/lib/main.dart" from="return MaterialApp" to="}"?>
-```diff
---- step2/lib/main.dart
-+++ step3/lib/main.dart
-@@ -48,3 +59,3 @@
-     return MaterialApp(
-       title: 'Flutter layout demo',
-       home: Scaffold(
-@@ -54,8 +65,9 @@
-         body: Column(
-           children: [
-             titleSection,
-+            buttonSection,
-           ],
-         ),
-       ),
-     );
-   }
+class BuildButtonColumn extends StatelessWidget {
+  const BuildButtonColumn({
+    super.key,
+    required this.color,
+    required this.icon,
+    required this.label,
+  });
+
+  final Color color;
+  final IconData icon;
+  final String label;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+// ···
+    );
+  }
+}
 ```
 
-## Step 4: Implement the text section
+### Update the app to display the button section
+
+Add the button section to the `children` array.
+
+    <?code-excerpt path-base="layout/lakes"?>
+    
+    <?code-excerpt "step{2,3}/lib/main.dart (addWidget)" title?>
+    ```diff
+    --- step2/lib/main.dart (addWidget)
+    +++ step3/lib/main.dart (addWidget)
+    @@ -5,6 +5,7 @@
+             name: 'Oeschinen Lake Campground',
+             location: 'Kandersteg, Switzerland',
+           ),
+    +      ButtonSection(),
+         ],
+       ),
+     ),
+    ```
+
+## Add the Text section
+
+In this section, add the text description to this app.
```

Comment:
As a reader, I was a bit confused reading this, as it made it seem like I should go off and do that myself with no help. Perhaps be more explicit with "In the following section" or "In the following steps"?